polyfill/test/duration.mjs

```diff
@@ -685,6 +685,18 @@ describe('Duration', () => {
       throws(() => oneDay.add(hours24, { relativeTo: { year: 2019, month: 11 } }), TypeError);
       throws(() => oneDay.add(hours24, { relativeTo: { year: 2019, day: 3 } }), TypeError);
     });
+    it('throws with invalid offset in relativeTo', () => {
+      throws(
+        () =>
+          Temporal.Duration.from('P2D').add('P1M', {
+            relativeTo: { year: 2022, month: 11, day: 26, offset: '+088:00', timeZone: 'Asia/Tokyo' }
+          }),
+        {
+          name: 'RangeError',
+          message: 'Invalid offset string'
+        }
```

Comment:
```suggestion
        RangeError
```
I'd prefer not to assert anything about the error message, since the spec doesn't mandate any particular message. (comment applies to all the tests)

Reply:
Normally I strongly agree, but in this case the RangeError was being thrown in the wrong place. If the code is working correctly, then it will throw a RangeError "early" (because the offset is bad) while if it's broken then it will assume that an invalid offset means `'+00:00'` and the RangeError is only thrown because the offset doesn't match the time zone. This would hide the bug.

I guess an alternative would be to use an IANA time zone that is legitimately `+00:00` at that exact time, and validate that a RangeError is thrown (without checking the message). This would let me simplify the tests. I'll plan on doing that unless you think I should do something else.

Reply:
I agree 100%, it would be better to run a test case that does not throw with the buggy code, and throw after the bug is fixed.

Reply:
Done.